hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1294,17 +1295,17 @@ private OzoneFileStatus getOzoneFileStatusFSO(String volumeName,
 
     if (fileStatus != null) {
       // if the key is a file then do refresh pipeline info in OM by asking SCM
-      if (fileStatus.isFile()) {
+      if (fileStatus.isFile() && !args.isHeadOp()) {
         OmKeyInfo fileKeyInfo = fileStatus.getKeyInfo();
-        if (latestLocationVersion) {
+        if (args.getLatestVersionLocation()) {
```

Comment:
@adoroszlai, same here as well. Could you please keep only refresh() and sortDatanodes() inside `if (!args.isHeadOp())` block and move LatestVersionLocation outside.